framework/src/st_things/things_stack/things_stack.c

```diff
@@ -194,6 +220,12 @@ void things_wifi_sta_connected(wifi_manager_result_e res)
 	}
 	THINGS_LOG_D(THINGS_INFO, TAG, "T%d --> %s", getpid(), __FUNCTION__);
 
+#ifdef CONFIG_NETUTILS_NTPCLIENT
+	sync_time_from_ntp();
```

Comment:
if things_wifi_sta_connected is the callback from wifi_manager then it would be better to call 'sync_time_from_ntp()' in other place. you should return the callback asap because it is called from wpa_supplicant task.
wpa supplicant task would be delayed.